ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -60,11 +67,126 @@ const uint64_t partSize = 8 * 1024 * 1024; /* 8 MB. */
 /* Setting SDK options. */
 Aws::SDKOptions options;
 
+static int S3GetDirectory(const std::string &, const std::string &, bool, std::string &);
+static int S3CacheExists(WT_FILE_SYSTEM *, const std::string &, bool &);
+static int S3CachePath(WT_FILE_SYSTEM *, const std::string &, std::string &);
+static int S3Path(const std::string &, const std::string &, std::string &);
+static int S3Exist(WT_FILE_SYSTEM *, WT_SESSION *, const char *, bool *);
 static int S3CustomizeFileSystem(
   WT_STORAGE_SOURCE *, WT_SESSION *, const char *, const char *, const char *, WT_FILE_SYSTEM **);
 static int S3AddReference(WT_STORAGE_SOURCE *);
 static int S3FileSystemTerminate(WT_FILE_SYSTEM *, WT_SESSION *);
 
+/*
+ * S3Exist --
+ *     Return if the file exists. First checks the cache, and then the S3 Bucket.
+ */
+static int
+S3Exist(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session, const char *name, bool *existp)
+{
+    S3_STORAGE *s3;
+    int ret;
+    *existp = false;
+    s3 = FS2S3(fileSystem);
+    S3_FILE_SYSTEM *fs = (S3_FILE_SYSTEM *)fileSystem;
+    bool exists;
+
+    /* First check to see if the file exists in the cache. */
+    if ((ret = S3CacheExists(fileSystem, name, exists)) != 0)
+        return (ret);
+
+    /* It's not in the cache, try the s3 bucket. */
+    if (!exists)
+        ret = fs->conn->ObjectExists(fs->bucketName, name, exists);
+
+    *existp = exists;
+    return (ret);
+}
+
+/*
+ * S3Path --
+ *     Construct a pathname from the directory and the object name.
+ */
+static int
+S3Path(const std::string &dir, const std::string &name, std::string &path)
```

Comment:
This function only returns a 0 ever, so there are no failure cases. Why not make the return type the path itself.